tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+                    // error handling for infinity pub key
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let message_vec: Vec<u8> = decode_hex_with_prefix(&self.input.message);
+        let message: &[u8] = message_vec.as_ref();
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sk) => sk,
+                Err(_) => {
+                    // this is the case where the tampered signature cannot be read in as real signature
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let verify_result = signature.verify(&pubkey, message);
+        verify_result == self.output
+    }
+}
+
+#[test]
+#[cfg(feature = "ef_spec_tests")]
+fn test_fast_aggregate_verify() {
+    for entry in
+        glob("consensus-spec-tests/tests/general/phase0/bls/fast_aggregate_verify/small/**/*.yaml")
+            .expect("Failed to read glob pattern")
+    {
+        match entry {
+            Ok(path) => {
+                println!("{:?}", path.display());
+                let file = File::open(path).expect("file exists");
+                let test_case: FastAggVerifyTestIO =
+                    serde_yaml::from_reader(file).expect("is well-formatted yaml");
+                assert!(test_case.fast_aggregate_verify())
+            }
+            Err(e) => println!("{:?}", e),
+        }
+    }
+}
+
+#[test]
+#[cfg(feature = "ef_spec_tests")]
+fn test_verify() {
+    for entry in glob("consensus-spec-tests/tests/general/phase0/bls/verify/small/**/*.yaml")
+        .expect("Failed to read glob pattern")
+    {
+        match entry {
+            Ok(path) => {
+                println!("{:?}", path.display());
+                let file = File::open(path).expect("file exists");
+                let test_case: VerifyTestIO =
+                    serde_yaml::from_reader(file).expect("is well-formatted yaml");
+                assert!(test_case.verify())
+            }
+            Err(e) => println!("{:?}", e),
+        }
+    }
+}
+#[test]
+#[cfg(feature = "ef_spec_tests")]
+fn test_aggregate_verify() {
+    for entry in
+        glob("consensus-spec-tests/tests/general/phase0/bls/aggregate_verify/small/**/*.yaml")
+            .expect("Failed to read glob pattern")
+    {
+        match entry {
+            Ok(path) => {
+                println!("{:?}", path.display());
+                let file = File::open(path).expect("file exists");
+                let test_case: AggVerifyTestIO =
+                    serde_yaml::from_reader(file).expect("is well-formatted yaml");
+                assert!(test_case.aggregate_verify())
```

Comment:
a minor thing but if you want to go the route of attaching the verification logic inside some "verify" method of some `FooTestIO` struct, i'd consider just calling the method `verify` or `execute` or something

then you can more easily copy/paste this code to use in multiple places -- may save you some work here

Reply:
if you moved to something like `execute`, i'd actually change the type signature to `() -> ()` so that you can drop the `assert!` inline here.

then up above in each test case, you can `assert!` accordingly

i think it is a weak idiom to get some computed result and `assert_eq!` against the expected/provided result so it may seem a bit more at home to a rust dev seeing this code if you follow the same pattern